EpikCord/__init__.py

```diff
@@ -1216,6 +1216,32 @@ def option_autocomplete(self, option_name: str):
         def wrapper(func):
             self.autocomplete_options[option_name] = func
         return wrapper
+    def checks(self, *args):
+      
+        """Checks the conditions given to it
+        Parameters:
+            All parameters should be functions returning boolean
+        
+        Returns:
+            A wrapper for a function
+        """
+        async def wrap(func:Callable):
+            check_no = len(args)
+            success_check = 0
+            for checker in args:
+                checker:Callable = checker
+                checked:bool = checker()
```

Comment:
Add support for coroutines, maybe ?

Reply:
Couroutine checks??? what!

Reply:
Ah, hold on, for moment i thought saw "async def", but yes, how about supporting coroutines for checks ? Since maybe you need to do DB fetch, w/e

Reply:
they are just gonna do 
```py
def if_admin():
   ...
@check(if_admin())
```

Reply:
not 
```py
async def if_admin():
    ...
@check(if_admin())```

Reply:
it just adds an unnecessary word

Reply:
I see, but what are you gonna pass for check parameters ?

Reply:
hmm lemme experiment

Reply:
> not
> 
> ```python
> async def if_admin():
>     ...
> @check(if_admin())
> ```

A check *is* to check if the command can execute, so if you're checking if they're in a database of blacklisted users and to fetch you need to call another coroutine... that's an issue

Reply:
Thats what i just said ^

Reply:
ok
thats easy to implement

Reply:
You didn't write whole code anyway, so you can just add `async` to `def check` then await the `checker()`